test/unit/serialization/compact/CompactTest.js

```diff
@@ -244,4 +246,18 @@ describe('CompactTest', function () {
         const object = await serializationService.toObject(data);
         object.should.be.deep.equal(employer);
     });
+
+    it('should not serialize without nested fields serializer', async function() {
+        const bundle = createSerializationService(
+            [new MainDTOSerializer(), new InnerDTOSerializer()]
+        );
+
+        serializationService = bundle.serializationService;
+
+        const mainDTO = createMainDTO();
+        await expect(serialize(
```

Comment:
Can you assert the exception message contains "No serializer is registered" (or similar) here? 

So you can use any string that will make sure newly added error is thrown. A random `HazelcastSerializationError` thrown will make this test pass.

Reply:
Fixed at https://github.com/hazelcast/hazelcast-nodejs-client/pull/1365/commits/3345e5697bb3247845774bd13ae536ebe3124fc2.